src/app/submit.service.ts

```diff
@@ -23,7 +23,15 @@ export class SubmitService {
       withLatestFrom(this.data$),
       mergeMap(([_, data]) => this.http.patch<OpenBabelData>(this.submitUrl, data).pipe(
         tap((result) => this.outputSubject.next(result))
-      ))
+      )),
+      catchError(this.handleError<OpenBabelData>('submit', this.blankData))
     );
   }
+
```

Comment:
Could you start adding in some doc comments, explaining purpose of this method, what it's suppose to handle and when it might get called?